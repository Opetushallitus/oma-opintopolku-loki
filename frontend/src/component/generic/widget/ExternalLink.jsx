import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import ExternalLinkIcon from 'icon/external-link-icon.svg'
import { Bold, Link } from 'ui/typography'

const PositionedLinkIcon = styled(ExternalLinkIcon)`
  padding: 0 0.25rem;
  position: relative;
  top: 2px;
`

const ExternalLink = ({ text, url, openInNewTab = false }) => (
  <Link target={openInNewTab ? '_blank' : undefined} href={url}>
    <PositionedLinkIcon/>
    <Bold>{text || url }</Bold>
  </Link>
)

ExternalLink.propTypes = {
  text: PropTypes.string,
  url: PropTypes.string.isRequired,
  openInNewTab: PropTypes.bool
}

export default ExternalLink
