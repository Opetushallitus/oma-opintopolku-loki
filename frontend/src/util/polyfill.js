import 'core-js/features/promise'
import 'core-js/features/string/starts-with'
