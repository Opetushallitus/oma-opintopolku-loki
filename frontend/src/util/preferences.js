import Cookies from 'js-cookie'

export const lang = Cookies.get('lang') || 'fi'
