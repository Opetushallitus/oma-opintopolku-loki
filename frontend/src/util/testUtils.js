/* istanbul ignore file */
import renderer from 'react-test-renderer'

export function matchesSnapshot (component) {
  const renderedComponent = renderer.create(component)
  const componentJSON = renderedComponent.toJSON()
  expect(componentJSON).toMatchSnapshot()
  renderedComponent.unmount()
}
