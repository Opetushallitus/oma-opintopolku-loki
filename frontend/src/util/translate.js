import Cookies from 'js-cookie'

export default key => {
  const lang = Cookies.get('lang') || 'fi'
  const translations = window.translationsMap || {}
  const versions = translations[key] || {}
  const translated = versions[lang]

  if (!translated) {
    console.error(`Translation missing for language ${lang}: ${key}`)
    return key
  }

  return translated
}
