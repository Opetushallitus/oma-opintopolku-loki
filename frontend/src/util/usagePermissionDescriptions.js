import t from 'util/translate'

import permissionsMap from 'Resources/mapping/usagePermissionDescriptions'

export const getTranslatedUsagePermissionDescription = organizationOid => {
  const permission = permissionsMap[organizationOid] || permissionsMap.default
  return t(permission)
}

export const isMydataPartner = organizationOid => {
  return Boolean(permissionsMap[organizationOid])
}
