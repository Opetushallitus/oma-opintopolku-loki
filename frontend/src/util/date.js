import moment from 'moment'
import { lang } from 'util/preferences'

moment.locale(lang)

export const isoStringToDate = isoString => moment(isoString).format('l, LT')
