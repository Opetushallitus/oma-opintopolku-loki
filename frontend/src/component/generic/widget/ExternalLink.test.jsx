import React from 'react'
import ExternalLink from './ExternalLink'
import { matchesSnapshot } from 'util/testUtils'

describe('ExternalLink', () => {
  it('should render external link', () => {
    matchesSnapshot(<ExternalLink
      text={'Teksti'}
      url={'http://localhost'}
    />)
  })

  it('should render external link that targets new tab', () => {
    matchesSnapshot(<ExternalLink
      text={'Teksti'}
      url={'http://localhost'}
      openInNewTab={true}
    />)
  })

  it('should render link without explicit text', () => {
    matchesSnapshot(<ExternalLink
      url={'http://localhost'}
    />)
  })
})
