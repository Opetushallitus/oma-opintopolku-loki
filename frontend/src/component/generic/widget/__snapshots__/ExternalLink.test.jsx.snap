// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExternalLink should render external link 1`] = `
<a
  className="typography__Link-sc-1yq3b4j-5 epyGqB"
  href="http://localhost"
>
  <test-file-stub
    className="ExternalLink__PositionedLinkIcon-sc-1in9m5g-0 fdjWcT"
  />
  <b
    className="typography__Bold-sc-1yq3b4j-6 jaOKTJ"
  >
    Teksti
  </b>
</a>
`;

exports[`ExternalLink should render external link that targets new tab 1`] = `
<a
  className="typography__Link-sc-1yq3b4j-5 epyGqB"
  href="http://localhost"
  target="_blank"
>
  <test-file-stub
    className="ExternalLink__PositionedLinkIcon-sc-1in9m5g-0 fdjWcT"
  />
  <b
    className="typography__Bold-sc-1yq3b4j-6 jaOKTJ"
  >
    Teksti
  </b>
</a>
`;

exports[`ExternalLink should render link without explicit text 1`] = `
<a
  className="typography__Link-sc-1yq3b4j-5 epyGqB"
  href="http://localhost"
>
  <test-file-stub
    className="ExternalLink__PositionedLinkIcon-sc-1in9m5g-0 fdjWcT"
  />
  <b
    className="typography__Bold-sc-1yq3b4j-6 jaOKTJ"
  >
    http://localhost
  </b>
</a>
`;
